### PR TITLE
I3S-picking - refactor getting attributes from selected feature

### PR DESCRIPTION
**Pull request by dryabinin-actionengine**

### CODE REVIEW

**Review by ibgreen — APPROVED** [review]
This is a lot of code we are adding to the tile layer.

Ultimate we are going to want to merge our changes back to deck.gl.

I would prefer if as much data normalization as possible was done in the I3SLoader so that the TileLayer can be small.

**Comment by dryabinin-actionengine** [comment]
> This is a lot of code we are adding to the tile layer.
> 
> Ultimate we are going to want to merge our changes back to deck.gl.
> 
> I would prefer if as much data normalization as possible was done in the I3SLoader so that the TileLayer can be small.

Moved this logic to the `I3SAttributeLoader`. I think make sense to do this code as a part of `I3SAttributeLoader` because we are working with attributes. Also added tests there.

File: modules/i3s/src/i3s-attribute-loader.js [hunk 1]
```diff
@@ -25,3 +26,61 @@ async function parse(data, options) {
...
+    attributeStorageInfo,
+    layerFeaturesAttributes,
+    calculatedFeatureIndex
+  );
+}
+
+function getFeatureAttributeIndex(layerFeaturesAttributes, featureId) {
+  let featureIndex = -1;
+
+  for (let index = 0; index < layerFeaturesAttributes.length; index++) {
```
Comment (ibgreen) [inline modules/i3s/src/i3s-attribute-loader.js:57]: When we load the tile, perhaps we could just create a map `attributesByObjectId` in the generated tile so the application can do lookup without calling a function?

**Review by ibgreen — APPROVED** [review]

File: modules/i3s/src/i3s-attribute-loader.js (hunk 1 above)
Comment (dryabinin-actionengine) [reply modules/i3s/src/i3s-attribute-loader.js:57]: Will do it in further PR

**Pull request merged**